File: Aula_0420/Ex04.py
class ContaBancaria:
  def __init__(self, titular, numConta):
    self.__titular = "sem nome"
    self.__numConta = "sem número"
    self.__saldo = 0
    if titular != "": self.__titular = titular
    if numConta != "": self.__numConta = numConta  
  def SetTitular(self, titular): 
    if titular != "": self.__titular = titular
  def SetNumConta(self, numConta): 
    if numConta != "": self.__numConta = numConta
  def Depositar(self, v): 
    if v < 0: raise ValueError() 
    if v > 0: self.__saldo += v
  def Sacar(self, v):
    if v < 0: raise ValueError() 
    if v > 0 and v <= self.__saldo:
      self.__saldo -= v
      return True
    return False
  def GetTitular(self):
    return self.__titular
  def GetNumConta(self):
    return self.__numConta
  def GetSaldo(self):
    return self.__saldo
  def __str__(self):
    return f"{self.__titular} {self.__numConta} {self.__saldo:.2f}"

x = ContaBancaria("Gilbert", "1234")
print(x.GetTitular())
print(x.GetNumConta())
print(x.GetSaldo())
x.Depositar(1000)
print(x.GetSaldo())
if (x.Sacar(1400)): print(x.GetSaldo())
else: print("Saldo insuficiente")
print(x);

File: Aula_0420/Ex04.cs
using System;
class Program {
  public static void Main() {
    ContaBancaria x = new ContaBancaria("Gilbert", "1234");
    //ContaBancaria x = new ContaBancaria();
    //x.SetTitular("Gilbert");
    //x.SetNumConta("1234");
    Console.WriteLine(x.GetTitular());
    Console.WriteLine(x.GetNumConta());
    Console.WriteLine(x.GetSaldo());
    x.Depositar(1000);
    Console.WriteLine(x.GetSaldo());
    if (x.Sacar(1400)) Console.WriteLine(x.GetSaldo());
    else Console.WriteLine("Saldo insuficiente");
    Console.WriteLine(x);
  }
}
class ContaBancaria {
  private string titular = "sem nome";
  private string numConta = "sem número";
  private double saldo;
  public ContaBancaria() { }
  public ContaBancaria(string t, string n) {
    if (t != "") titular = t;
    SetNumConta(n);
  }
  public void SetTitular(string titular) { 
    if (titular != "") this.titular = titular; 
  }
  public void SetNumConta(string numConta) { 
    if (numConta != "") this.numConta = numConta;
  }
  public void Depositar(double v) { 
    if (v < 0) throw new ArgumentOutOfRangeException();
    if (v > 0) saldo += v; 
  }
  public bool Sacar(double v) { 
    if (v < 0) throw new ArgumentOutOfRangeException();
    if (v > 0 && v <= saldo) { saldo -= v; return true; }
    return false;
  }
  public string GetTitular() { return titular; }
  public string GetNumConta() { return numConta; }
  public double GetSaldo() { return saldo; }
  public override string ToString() {
    return $"{titular} {numConta} {saldo:f2}";
  }
}